dts/x86/arduino_101.dts

```diff
@@ -0,0 +1,29 @@
+/dts-v1/;
+
+#include "intel_curie.dtsi"
+
+/ {
+	model = "Arduino 101";
+	compatible = "intel,quark";
+
+	aliases {
+		uart_0 = &uart0;
+		uart_1 = &uart1;
+	};
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,console = &uart1;
+	};
+};
+
+&uart0 {
+	status = "ok";
+	current-speed = <1000000>;
```

Comment:
yes HW flowcontrol is comming from Kconfig. DTS support of  HW flowcontrol requires changes in  dts/common/yaml/uart.yaml so i will push those changes as new patch.